src/Advection/vector_invariant_upwinding.jl

```diff
@@ -0,0 +1,154 @@
+Contrarily, only the Kinetic Energy gradient term corresponding to the transporting velocity is upwinded.
+
+Keyword arguments
+=================  
+
+- `cross_scheme`: Advection scheme used for cross-reconstructed terms (tangential velocities) 
+                  in the kinetic energy gradient. Defaults to `CenteredSecondOrder()`.
+- `divergence_stencil`: Stencil used for smoothness indicators of `δx_U + δy_V` in case of a 
+                        `WENO` upwind reconstruction. Defaults to `DefaultStencil()`.
+- `δu²_stencil`: Stencil used for smoothness indicators of `δx_u²` in case of a `WENO` upwind reconstruction. 
+                 Defaults to `FunctionStencil(u_smoothness)` 
+- `δv²_stencil`: Same as `δu²_stencil` but for the smoothness of `δy_v²`
+                 Defaults to `FunctionStencil(v_smoothness)`
+"""
+CrossAndSelfUpwinding(; cross_scheme       = CenteredSecondOrder(),
+                        divergence_stencil = DefaultStencil(),
+                        δu²_stencil        = FunctionStencil(u_smoothness),
+                        δv²_stencil        = FunctionStencil(v_smoothness),
+                        ) = CrossAndSelfUpwinding(extract_centered_scheme(cross_scheme), divergence_stencil, δu²_stencil, δv²_stencil)
+
+"""
+    VelocityUpwinding(; cross_scheme = CenteredSecondOrder()) = VelocityUpwinding(cross_scheme)
+                                
+Upwinding treatment for Divergence fluxes and Kinetic Energy gradient in the Vector Invariant formulation, whereas only 
+the terms corresponding to the transporting velocity are upwinded. (i.e., terms in `u` in the zonal momentum equation and 
+terms in `v` in the meridional momentum equation). Contrarily to `OnlySelfUpwinding`, the reconstruction (and hence the
+upwinding) is done _inside_ the gradient operator, i.e., velocities are reconstructed instead of velocity derivatives.
+
+Keyword arguments
+=================  
+
+- `cross_scheme`: Advection scheme used for cross-reconstructed terms (tangential velocities) 
+                    in the kinetic energy gradient and the divergence flux. Defaults to `CenteredSecondOrder()`.
+"""
+VelocityUpwinding(; cross_scheme = CenteredSecondOrder()) = VelocityUpwinding(extract_centered_scheme(cross_scheme))
+                    
+Base.summary(a::OnlySelfUpwinding)     = "OnlySelfUpwinding"
+Base.summary(a::CrossAndSelfUpwinding) = "CrossAndSelfUpwinding"
+Base.summary(a::VelocityUpwinding)     = "VelocityUpwinding"
+
+Base.show(io::IO, a::OnlySelfUpwinding) =
+    print(io, summary(a), " \n",
+            " KE gradient and Divergence flux cross terms reconstruction: ", "\n",
+            " └── $(summary(a.cross_scheme))", "\n",
+            " Smoothness measures: ", "\n",
+            " ├── smoothness δU: $(a.δU_stencil)", "\n", 
+            " ├── smoothness δV: $(a.δV_stencil)", "\n",
+            " ├── smoothness δu²: $(a.δu²_stencil)", "\n",
+            " └── smoothness δv²: $(a.δv²_stencil)")
```

Comment:
```suggestion
            " ├── δU: $(a.δU_stencil)", "\n", 
            " ├── δV: $(a.δV_stencil)", "\n",
            " ├── δu²: $(a.δu²_stencil)", "\n",
            " └── δv²: $(a.δv²_stencil)")
```

Reply:
hmmm, not sure about this. It might look like the stencils are for calculating the actual `δU`, `δV`, `δu ²`, `δu ²` while they only calculating the smoothness of those quantities